feat: add `stats/base/ndarray/dcovarmtk`

gururaj1512 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   feat: add `stats/base/ndarray/dcovarmtk`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gururaj1512 committed:
feat: add `stats/base/ndarray/dcovarmtk`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
Update README.md

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 committed:
Update main.js

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 committed:
Update index.d.ts

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

gururaj1512 commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+
+# dcovarmtk
+
+> Calculate the [covariance][covariance] of two one-dimensional double-precision floating-point ndarrays provided known means and using a one-pass textbook algorithm.
```
Is this definition correct?

kgryte replied:
Yes, this looks fine to me.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+var meany = scalar2ndarray( 1.0/3.0, opts );
+var correction = scalar2ndarray( 1.0, opts );
+
+var v = dcovarmtk( [ x, y, meanx, meany, correction ] );
```
@gururaj1512 Would you mind reordering the ndarrays here to

```
[ x, y, correction, meanx, meany ]
```

?

This ensures that the relative order of the ndarrays matches the order of the arguments in the corresponding strided array function

```
dcovarmtk( N, correction, meanx, x, strideX, meany, y, strideY )
```

Notice: `correction`, `meanx`, and `meany` order.

That applies throughout this PR.

gururaj1512 committed:
refactor: update parameter order

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+    2.  second one-dimensional input ndarray.
+    3.  a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [covariance][covariance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population [covariance][covariance], setting this parameter to `0` is t ...
+    4.  a zero-dimensional ndarray specifying mean of the first one-dimensional ndarray.
+    5.  a zero-dimensional ndarray specifying mean of the second one-dimensional ndarray.
```
```suggestion
    4.  a zero-dimensional ndarray specifying the mean of the first one-dimensional ndarray.
    5.  a zero-dimensional ndarray specifying the mean of the second one-dimensional ndarray.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+
+    1.  first one-dimensional input ndarray.
+    2.  second one-dimensional input ndarray.
+    3.  a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [covariance][covariance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population [covariance][covariance], setting this parameter to `0` is t ...
```
```suggestion
    3.  a zero-dimensional ndarray specifying the degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the [covariance][covariance] according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment and `N` corresponds to the number of elements in each input ndarray. When computing the population [covariance][covariance], setting this parameter to `0` is the standard choice (i.e., the provided arrays contain data constituting entire populations). When computing the unbiased sample [covariance][covariance], setting this parameter to `1` is the standard choice (i.e., the provided arrays contain data sampled from larger populations; this is commonly referred to as Bessel's correction).
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+<section class="notes">
+
+## Notes
+
+-   If provided an empty one-dimensional ndarray, the function returns `NaN`.
```
```suggestion

-   Both input ndarrays should have the same number of elements.
-   If provided empty one-dimensional ndarrays, the function returns `NaN`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/README.md:
```diff
@@ -0,0 +1,216 @@
+    'dtype': 'float64'
+};
+
+// Create first one-dimensional ndarray containing pseudorandom integers drawn from a discrete uniform distribution:
+var xbuf = discreteUniform( 10, -50, 50, opts );
+var x = new ndarray( opts.dtype, xbuf, [ xbuf.length ], [ 1 ], 0, 'row-major' );
+console.log( ndarray2array( x ) );
+
+// Create second one-dimensional ndarray containing pseudorandom integers drawn from a discrete uniform distribution:
```
```suggestion
// Create one-dimensional ndarrays containing pseudorandom numbers:
var xbuf = discreteUniform( 10, -50, 50, opts );
var x = new ndarray( opts.dtype, xbuf, [ xbuf.length ], [ 1 ], 0, 'row-major' );
console.log( ndarray2array( x ) );

```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/repl.txt:
```diff
@@ -0,0 +1,64 @@
+
+{{alias}}( arrays )
+    Computes the covariance of two one-dimensional double-precision
+    floating-point ndarrays provided known means and using a one-pass textbook
+    algorithm.
```
```suggestion
    Computes the covariance of two one-dimensional double-precision floating-
    point ndarrays provided known means and using a one-pass textbook algorithm.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/repl.txt:
```diff
@@ -0,0 +1,64 @@
+    floating-point ndarrays provided known means and using a one-pass textbook
+    algorithm.
+
+    If provided an empty ndarray, the function returns `NaN`.
```
Should update to match README notes.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/repl.txt:
```diff
@@ -0,0 +1,64 @@
+        -   a zero-dimensional ndarray specifying degrees of freedom
+        adjustment. Setting this parameter to a value other than `0` has the
+        effect of adjusting the divisor during the calculation of the
+        covariance according to `N-c` where `c` corresponds to the provided
```
Should be updated to match README suggestion.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/repl.txt:
```diff
@@ -0,0 +1,64 @@
+        -   a zero-dimensional ndarray specifying mean of the first
+        one-dimensional ndarray.
+        -   a zero-dimensional ndarray specifying mean of the second
+        one-dimensional ndarray.
```
```suggestion
        -   a zero-dimensional ndarray specifying the mean of the first one-
        dimensional ndarray.
        -   a zero-dimensional ndarray specifying the mean of the second one-
        dimensional ndarray.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/types/index.d.ts:
```diff
@@ -0,0 +1,68 @@
+*
+*     -   first one-dimensional input ndarray.
+*     -   second one-dimensional input ndarray.
+*     -   a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the covariance according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population covariance, setting this parameter to `0` is the standard choice (i.e.,  ...
```
Should update to match README suggestion.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/docs/types/index.d.ts:
```diff
@@ -0,0 +1,68 @@
+*     -   second one-dimensional input ndarray.
+*     -   a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the covariance according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population covariance, setting this parameter to `0` is the standard choice (i.e.,  ...
+*     -   a zero-dimensional ndarray specifying mean of the first one-dimensional ndarray.
+*     -   a zero-dimensional ndarray specifying mean of the second one-dimensional ndarray.
```
```suggestion
*     -   a zero-dimensional ndarray specifying the mean of the first one-dimensional ndarray.
*     -   a zero-dimensional ndarray specifying the mean of the second one-dimensional ndarray.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/examples/index.js:
```diff
@@ -0,0 +1,51 @@
+	'dtype': 'float64'
+};
+
+// Create first one-dimensional ndarray containing pseudorandom integers drawn from a discrete uniform distribution:
```
Same changes as README.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/lib/main.js:
```diff
@@ -0,0 +1,91 @@
+*
+*     -   first one-dimensional input ndarray.
+*     -   second one-dimensional input ndarray.
+*     -   a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the covariance according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population covariance, setting this parameter to `0` is the standard choice (i.e.,  ...
```
Should update to match README suggestion.

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/lib/main.js:
```diff
@@ -0,0 +1,91 @@
+*     -   second one-dimensional input ndarray.
+*     -   a zero-dimensional ndarray specifying degrees of freedom adjustment. Setting this parameter to a value other than `0` has the effect of adjusting the divisor during the calculation of the covariance according to `N-c` where `c` corresponds to the provided degrees of freedom adjustment. When computing the population covariance, setting this parameter to `0` is the standard choice (i.e.,  ...
+*     -   a zero-dimensional ndarray specifying mean of the first one-dimensional ndarray.
+*     -   a zero-dimensional ndarray specifying mean of the second one-dimensional ndarray.
```
```suggestion
*     -   a zero-dimensional ndarray specifying the mean of the first one-dimensional ndarray.
*     -   a zero-dimensional ndarray specifying the mean of the second one-dimensional ndarray.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/ndarray/dcovarmtk/package.json:
```diff
@@ -0,0 +1,67 @@
+{
+  "name": "@stdlib/stats/base/ndarray/dcovarmtk",
+  "version": "0.0.0",
+  "description": "Computes the covariance of two one-dimensional double-precision floating-point ndarrays provided known means and using a one-pass textbook algorithm.",
```
```suggestion
  "description": "Compute the covariance of two one-dimensional double-precision floating-point ndarrays provided known means and using a one-pass textbook algorithm.",
```

kgryte reviewed (changes requested):
Left a few nits. Once addressed, should be ready.

gururaj1512 committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM
